taichi/ir/ir.cpp

```diff
@@ -317,13 +317,22 @@ UnaryOpStmt::UnaryOpStmt(UnaryOpType op_type, Stmt *operand)
     : op_type(op_type), operand(operand) {
   TI_ASSERT(!operand->is<AllocaStmt>());
   cast_type = DataType::unknown;
-  cast_by_value = true;
   TI_STMT_REG_FIELDS;
 }
 
+bool UnaryOpStmt::is_cast() const {
+  switch (op_type) {
```

Comment:
I think we can just `return op_type == cast_value || op_type == cast_bits;` here -- `switch` looks like an overkill.

Reply:
A slightly further improvement: can we define a helper on `UnaryOpType` itself, e.g. `bool is_cast(UnaryOpType t);`, then we don't have to duplicate the logic in `UnaryOpStmt` and `UnaryOpExpr`.